jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/CloudFormationTemplate.kt

```diff
@@ -0,0 +1,83 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudformation
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiElement
+import org.jetbrains.yaml.YAMLFileType
+import org.jetbrains.yaml.YAMLLanguage
+import software.aws.toolkits.jetbrains.services.cloudformation.yaml.YamlCloudFormationTemplate
+import java.io.File
+
+interface CloudFormationTemplate {
+    fun resources(): Sequence<Resource>
+
+    fun getResourceByName(logicalName: String): Resource? {
```

Comment:
How do you feel about doing the resource type mapping here instead? We could have a registry that knows what type maps to what implementation (e.g `LambdaResource`, `FunctionResource` etc); if we can't find a specific type then return the base `Resource`

Thoughts?